dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/diagnostics/DiagnosticsRegistry.java

```diff
@@ -0,0 +1,38 @@
+package hu.elte.txtuml.api.model.execution.diagnostics;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import hu.elte.txtuml.api.model.ModelClass;
+import hu.elte.txtuml.api.model.StateMachine.Vertex;
+
+/**
+ * Class for storing the state of an execution's diagnostics.
+ * This state consists of active state-machine vertices assigned to instances of state-machine classes.
+ * 
+ * @author szokolai-mate
+ */
+public class DiagnosticsRegistry {
```

Comment:
Redundant and unsafe implementation, should be replaced by an existing thread safe map.